content/blog/2021/10/02/MonitorTheSolver.adoc

```diff
@@ -0,0 +1,209 @@
+In Quarkus, to connect OptaPlanner to https://prometheus.io/[Prometheus], you would add its extension as a dependency in the `pom.xml`:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-micrometer-registry-prometheus</artifactId>
+</dependency>
+----
+
+In Spring Boot, add the Spring Boot actuator to your project along with the registry dependency:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-actuator</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.micrometer</groupId>
+    <artifactId>micrometer-registry-prometheus</artifactId>
+</dependency>
+----
+and enable the metrics endpoint in `application.properties`:
+[source,properties]
+----
+management.endpoints.web.exposure.include=metrics,prometheus
+----
+
+For information on how to connect Micrometer to other monitoring systems, visit https://micrometer.io/docs[the Micrometer documentation].
+
+== Monitoring the Solver
+
+Now that we know what monitoring is and how it can be useful, let's walk through an actual example.
+We'll be running a modified version of the https://github.com/kiegroup/optaplanner-quickstarts/tree/stable/use-cases/school-timetabling[school timetabling quickstart] with support for multitenancy.
+You can find the complete source code, along with scripts for running it on the https://github.com/Christopher-Chianelli/optaplanner-micrometer-blog[optaplanner-micrometer-blog] GitHub page.
+
+== Starting the application
+
+Clone the example code:
+
+[source,bash]
+----
+git clone https://github.com/Christopher-Chianelli/optaplanner-micrometer-blog
+cd optaplanner-micrometer-blog
+----
+
+Start the application in development mode:
+
+[source,bash]
+----
+mvn quarkus:dev
+----
+
+After the application is started, you can access the application at http://localhost:8080.
+To see available metrics for the application, visit http://localhost:8080/q/metrics.
+
+image:ExampleMonitoringApp.png[Example Application]
+
+== Starting Prometheus
+
+Prometheus has a prebuilt Docker image on Dockerhub that can be used for running Prometheus.
+We'll need to modify its configuration so it'll scrape our metrics endpoint.
+
+Create the `prometheus.yml` file with the following text:
+
+[source,yaml]
+----
+scrape_configs:
+- job_name: local application
+  scrape_interval: 1s
+  metrics_path: /q/metrics
+  static_configs:
+  - targets:
+    - localhost:8080
+----
+
+This configures Prometheus to scrape metrics from `localhost:8080/q/metrics` every second.
+
+Start Prometheus with the preceding configuration:
+
+[source,bash]
+----
+docker run \
+    --network host \
+    --mount type=bind,source=prometheus.yml,destination=/etc/prometheus/prometheus.yml,ro=true,relabel=shared \
+     prom/prometheus
+----
+
+You can see the Prometheus UI by visiting http://localhost:9090.
+
+image::PrometheusUI.png[Prometheus UI]
```

Comment:
Can the prometheus UI image show an optaplanner metric?
Do we first need to explain monitoring with a non-optaplanner metric or can we directly explain it with an optaplanner metric?